src/System/Relay/middleware/cacheHeaderMiddleware.ts

```diff
@@ -54,6 +55,9 @@ export const cacheHeaderMiddleware = (props?: CacheHeaderMiddlewareProps) => {
       const foundRoute = findRoutesByPath({ path: url ?? "" })[0]
 
       switch (true) {
+        case isRequestCacheable(req): {
```

Comment:
we've got [this bit](https://github.com/artsy/force/blob/9592017a8127af37f309210fcd905c0ff01ac950/src/System/Relay/middleware/cacheHeaderMiddleware.ts#L19-L29), so thinking we should check a few conditions (if i'm understanding correctly)

Reply:
Yes but this boolean in isolation does need to be reusable b/c the `urlMiddleware` needs it to drop the access token header, so I chose to make it a separate condition and include this higher up in the switch here - it seemed easier to me. idk!

Reply:
I guess that method can check this and `return true` at the top.

Reply:
Imaging tho on a route or internal to a query in a route:
```tsx
{ 
  path: '/jobs'
  serverCacheTTL: 0,
  query: graphql`
    query jobsRouteQuery {
      jobs @cacheable {
        //
      }
    }
  `
}
```
Because of that route-level config, this should never cache. Same with components with `force: true`. Are those things properly validated here?

Reply:
The `@cacheable` directive is on the query-level.

There is no validations here (as you can see from the simple code!). If this directive exists on a query, it can be cached at the CDN. There is no interplay between the cache TTL (yet).

Reply:
ie - in your example, it's just nonsensical / dev error to include a directive that says this is cacheable, _and_ some sort-of custom config to have this not cacheable.

I think we can improve that (validation?) - but I'm happy to iterate on it, we don't need to solve every problem here.

Reply:
It seems non-sensical right now, but at scale its not. If we've got clear general rules around cache behavior 
- force: true
- serverCacheTTL: 0

it should be respected globally. And if that condition can be trivially checked here in this code with just a couple more lines, we should 🤷

Reply:
Updated - it was a little bit more than a couple of lines.